ansible/playbooks/AdoptOpenJDK_AIX_Playbook/roles/xlc_v13/tasks/main.yml

```diff
@@ -13,67 +13,85 @@
+      when: xlc13_images is defined and xlc13_images.rc == 0
+      tags: xlc13
 
-- debug: msg='Erorrs from the previous installp command normal'
-  when: xlc13.stat.islnk is not defined
-  tags: xlc13
+    - debug: msg='Errors from the previous installp command normal'
+      when: result.xlc is defined
+      tags: xlc13
+
+    - name: Clean XLC13 tmp files
+      file:
+        path: /tmp/usr
+        state: absent
+      tags: xlc13
+      when: xlc13_images is defined
+  when: xlc13.stat.islnk is undefined
 
 - name: TestIBM XLC13
   command: /opt/IBM/xlC/13.1.3/bin/xlc -qversion
   register: xlc13_qversion
+  ignore_errors: yes
   tags: xlc13
+  changed_when: false
 
 - name: Display XLC13 qversion information
   debug:
     msg: "{{ xlc13_qversion.stdout }}"
+  when: xlc13_qversion.failed == false
   tags: xlc13
+  changed_when: false
 
-- name: Symlink - /usr/bin/xlc
-  file: src=/opt/IBM/xlC/13.1.3/bin/xlc dest=/usr/bin/xlc state=link
-  when: xlc13.stat.islnk is not defined
-  tags: xlc13
+# Fail, set_fact, warn when xlc13_qversion.rc != 0
+- debug:
+    msg: "{{ xlc13_qversion.cmd }} returns: {{ xlc13_qversion.msg }}"
+  when: xlc13_qversion.failed == true
 
-- name: Symlink - /usr/bin/xlc++
-  file: src=/opt/IBM/xlC/13.1.3/bin/xlc++ dest=/usr/bin/xlc++ state=link
-  when: xlc13.stat.islnk is not defined
-  tags: xlc13
+# - name: Symlink - /usr/bin/xlc
+#   file: src=/opt/IBM/xlC/13.1.3/bin/xlc dest=/usr/bin/xlc state=link
+#   when: xlc13.stat.islnk is not defined
+#   tags: xlc13
 
-- name: Symlink - /usr/bin/xlC
-  file: src=/opt/IBM/xlC/13.1.3/bin/xlC dest=/usr/bin/xlC state=link
-  when: xlc13.stat.islnk is not defined
-  tags: xlc13
+# - name: Symlink - /usr/bin/xlc++
+#   file: src=/opt/IBM/xlC/13.1.3/bin/xlc++ dest=/usr/bin/xlc++ state=link
+#   when: xlc13.stat.islnk is not defined
+#   tags: xlc13
 
-- name: Symlink - /usr/bin/xlc_r
-  file: src=/opt/IBM/xlC/13.1.3/bin/xlc_r dest=/usr/bin/xlc_r state=link
-  when: xlc13.stat.islnk is not defined
-  tags: xlc13
+# - name: Symlink - /usr/bin/xlC
+#   file: src=/opt/IBM/xlC/13.1.3/bin/xlC dest=/usr/bin/xlC state=link
+#   when: xlc13.stat.islnk is not defined
+#   tags: xlc13
 
-- name: Symlink - /usr/bin/xlC_r
-  file: src=/opt/IBM/xlC/13.1.3/bin/xlc_r dest=/usr/bin/xlC_r state=link
-  when: xlc13.stat.islnk is not defined
-  tags: xlc13
+# - name: Symlink - /usr/bin/xlc_r
+#   file: src=/opt/IBM/xlC/13.1.3/bin/xlc_r dest=/usr/bin/xlc_r state=link
+#   when: xlc13.stat.islnk is not defined
+#   tags: xlc13
 
-- name: Symlink - /usr/bin/gxlC
-  file: src=/opt/IBM/xlC/13.1.3/bin/gxlC dest=/usr/bin/gxlC state=link
-  when: xlc13.stat.islnk is not defined
-  tags: xlc13
+# - name: Symlink - /usr/bin/xlC_r
+#   file: src=/opt/IBM/xlC/13.1.3/bin/xlc_r dest=/usr/bin/xlC_r state=link
+#   when: xlc13.stat.islnk is not defined
+#   tags: xlc13
 
-- name: Clean XLC13 tmp files
-  file:
-    path: /tmp/usr
-    state: absent
-  tags: xlc13
+# - name: Symlink - /usr/bin/gxlC
```

Comment:
Out of curiosity, why are these symlink tasks omitted? I would assume these get used somewhere by a process, but I am not entirely sure.

Reply:
It makes them the default. IMHO: better practice will be to set PATH for the compiler wanted (either in a .profile, but better in the script, or as a symlink in the $HOME/bin.

Remember - we are also installing the v16 compiler. Why does that not get symlinks?

So, I left them here - to have this discussion :)

Reply:
note also the comment: fail: or debug:

If you really want a playbook to stop dead in it's track - change debug: to fail: to be explicit about it.

Again, my preference will be to set a fact that summarizes what 'failed' aka is missing (the xlcv13 compiler) - and at the end of the playbook have a summary of what is needed to be fully prepared.

I really do not like stop and go (e.g., I use `make -i` to get an overview of what is broken rather than trial and trial and trial).

Reply:
> It makes them the default. IMHO: better practice will be to set PATH

I can understand that but there's no way I can accept this unless we are 100% sure that it wouldn't cause a build failure at the moment if the playbook was run on a new machine which didn't have the symlinks.

Reply:
I only commented them away - asif removing them. i.e., for existing situation - nothing changes, existing symlinks are not removed - AND - `missing` links are not placed so that users need to add something to PATH.

FYI: the original src: values are (mostly, if not all) symlinks themselves.

Reply:
Adding a comment:

As an example:
osuosl build-{1,2} seem to pre-date this change to the use of symbolic links (build-1 has none, build-2 has 2)
```
root@nim.bak:[/export/nim/resources]for i in 1 2 3 4 5 6^Jdo ^Jprint ojdk0$i^Jssh ojdk0$i.bak ls -l /usr/bin/xl*
> done
ojdk01
ls_64: cannot access '/usr/bin/xl*': A file or directory in the path name does not exist.
ojdk02
lrwxrwxrwx 1 root system 27 Jul  2  2020 /usr/bin/xlC -> /opt/IBM/xlC/13.1.3/bin/xlC
lrwxrwxrwx 1 root system 27 Jul  2  2020 /usr/bin/xlc -> /opt/IBM/xlc/13.1.3/bin/xlc
ojdk03
lrwxrwxrwx    1 root     system           27 Jul 29 2019  /usr/bin/xlC -> /opt/IBM/xlC/13.1.3/bin/xlC
lrwxrwxrwx    1 root     system           29 Jul 29 2019  /usr/bin/xlC_r -> /opt/IBM/xlC/13.1.3/bin/xlc_r
lrwxrwxrwx    1 root     system           27 Jul 29 2019  /usr/bin/xlc -> /opt/IBM/xlC/13.1.3/bin/xlc
lrwxrwxrwx    1 root     system           29 Jul 29 2019  /usr/bin/xlc++ -> /opt/IBM/xlC/13.1.3/bin/xlc++
lrwxrwxrwx    1 root     system           29 Jul 29 2019  /usr/bin/xlc_r -> /opt/IBM/xlC/13.1.3/bin/xlc_r
ojdk04
lrwxrwxrwx    1 root     system           27 Jan 28 2020  /usr/bin/xlC -> /opt/IBM/xlC/13.1.3/bin/xlC
lrwxrwxrwx    1 root     system           29 Jan 28 2020  /usr/bin/xlC_r -> /opt/IBM/xlC/13.1.3/bin/xlc_r
lrwxrwxrwx    1 root     system           27 Jan 28 2020  /usr/bin/xlc -> /opt/IBM/xlC/13.1.3/bin/xlc
lrwxrwxrwx    1 root     system           29 Jan 28 2020  /usr/bin/xlc++ -> /opt/IBM/xlC/13.1.3/bin/xlc++
lrwxrwxrwx    1 root     system           29 Jan 28 2020  /usr/bin/xlc_r -> /opt/IBM/xlC/13.1.3/bin/xlc_r
ojdk05
lrwxrwxrwx 1 root system 27 Sep 22 13:36 /usr/bin/xlC -> /opt/IBM/xlC/13.1.3/bin/xlC
lrwxrwxrwx 1 root system 29 Sep 22 13:37 /usr/bin/xlC_r -> /opt/IBM/xlC/13.1.3/bin/xlc_r
lrwxrwxrwx 1 root system 27 Sep 22 13:36 /usr/bin/xlc -> /opt/IBM/xlC/13.1.3/bin/xlc
lrwxrwxrwx 1 root system 29 Sep 22 13:36 /usr/bin/xlc++ -> /opt/IBM/xlC/13.1.3/bin/xlc++
lrwxrwxrwx 1 root system 29 Sep 22 13:37 /usr/bin/xlc_r -> /opt/IBM/xlC/13.1.3/bin/xlc_r
ojdk06
Warning: Permanently added 'ojdk06.bak' (ECDSA) to the list of known hosts.
lrwxrwxrwx 1 root system 27 Mar  8 19:54 /usr/bin/xlC -> /opt/IBM/xlC/13.1.3/bin/xlC
lrwxrwxrwx 1 root system 29 Mar  8 19:54 /usr/bin/xlC_r -> /opt/IBM/xlC/13.1.3/bin/xlc_r
lrwxrwxrwx 1 root system 27 Mar  8 19:54 /usr/bin/xlc -> /opt/IBM/xlC/13.1.3/bin/xlc
lrwxrwxrwx 1 root system 29 Mar  8 19:54 /usr/bin/xlc++ -> /opt/IBM/xlC/13.1.3/bin/xlc++
lrwxrwxrwx 1 root system 29 Mar  8 19:54 /usr/bin/xlc_r -> /opt/IBM/xlC/13.1.3/bin/xlc_r
```
* As far as PATH definition goes - all include `/opt/IBM/xlC/13.1.3/bin` in their PATH - but again - not consistent:
```
root@nim.bak:[/export/nim/resources]for i in 1 2 3 4 5 6^Jdo ^Jprint ojdk0$i^Jssh ojdk0$i.bak grep PATH= /etc/environment | grep -i xlc^Jdone
ojdk01
PATH=/opt/freeware/bin:/opt/IBM/xlC/13.1.3/bin:/usr/bin:/etc:/usr/sbin:/usr/ucb:/usr/bin/X11:/sbin:/usr/java6/jre/bin:/usr/java6/bin
ojdk02
PATH=/opt/freeware/bin:/opt/IBM/xlC/13.1.3/bin:/usr/bin:/etc:/usr/sbin:/usr/ucb:/usr/bin/X11:/sbin:/usr/java6/jre/bin:/usr/java6/bin
ojdk03
#PATH=/opt/freeware/bin:/opt/IBM/xlC/13.1.3/bin:/usr/bin:/etc:/usr/sbin:/usr/ucb:/usr/bin/X11:/sbin:/usr/java7_64/jre/bin:/usr/java7_64/bin
PATH=/opt/IBM/xlC/13.1.3/bin:/usr/bin:/etc:/usr/sbin:/usr/ucb:/usr/bin/X11:/sbin:/usr/java7_64/jre/bin:/usr/java7_64/bin
ojdk04
PATH=/opt/IBM/xlC/13.1.3/bin:/usr/bin:/etc:/usr/sbin:/usr/ucb:/usr/bin/X11:/sbin:/usr/java7_64/jre/bin:/usr/java7_64/bin
ojdk05
PATH=/opt/freeware/bin:/opt/IBM/xlC/13.1.3/bin:/usr/bin:/etc:/usr/sbin:/usr/ucb:/usr/bin/X11:/sbin:/usr/java6/jre/bin:/usr/java6/bin
ojdk06
PATH=/opt/freeware/bin:/opt/IBM/xlC/13.1.3/bin:/usr/bin:/etc:/usr/sbin:/usr/ucb:/usr/bin/X11:/sbin:/usr/java6/jre/bin:/usr/java6/bin:/opt/bin
```